### ISSUE TITLE
Can we save the composite protein after molecular docking instead of viewing it directly?

### ISSUE THREAD

**Issue by TuTuSklmq** [dockstring/dockstring#170]
I run dockstring on the SSH connection terminal, but the remote end cannot see the molecules very well after opening pymol, so I hope to save the products after docking the molecules and view them later on the local computer.
<img width="860" alt="微信图片_20240321113944" src="https://github.com/dockstring/dockstring/assets/155504943/971de4b9-1058-4f74-86b1-9fcf0c1ec846">

**Comment by AustinT** [comment]
Hi @TuTuSklmq , I think you can do one of two things:

## 1: use pickle

```python
from dockstring import load_target
import pickle

target = load_target('DRD2')
score, aux = target.dock(smiles)
with open("aux.pkl", "wb") as f:
  pickle.dump(aux, f)
```

Then, locally, with dockstring imported

```python
... same imports as above

target = load_target('DRD2')
with open("aux.pkl", "rb") as f:
  aux = pickle.load(f)

target.view([aux['ligand']])
```

## 2: forward X instance

If you run `ssh -X host` instead of `ssh host` it should allow you to open graphical applications on `host`, including `pymol`.

Google "X11 forwarding" for more instructions on how to handle this.

Hope this helps 😃

**Comment by TuTuSklmq** [comment]
I am glad to receive your reply, but I tried the first method and found that after saving as a pkl file, I cannot transfer the file to my local computer. It shows 'can not create local file', which is something I cannot solve.




ROSA&nbsp;&nbsp;&nbsp;MULTIFLORA
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dockstring/dockstring"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月21日(星期四) 晚上6:11
***@***.***&gt;;
抄送:&nbsp;"ROSA   ***@***.******@***.***&gt;;
主题:&nbsp;Re: [dockstring/dockstring] Can we save the composite protein after molecular docking instead of viewing it directly? (Issue #170)





 
Hi @TuTuSklmq , I think you can do one of two things:
 
1: use pickle
 from dockstring import load_target import pickle target = load_target('DRD2') score, aux = target.dock(smiles) with open("aux.pkl", "wb") as f:   pickle.dump(aux, f)
 
Then, locally, with dockstring imported
 ... same imports as above target = load_target('DRD2') with open("aux.pkl", "rb") as f:   aux = pickle.load(f) target.view([aux['ligand']])
 
2: forward X instance
 
If you run ssh -X host instead of ssh host it should allow you to open graphical applications on host, including pymol.
 
Google "X11 forwarding" for more instructions on how to handle this.
 
Hope this helps 😃

**Comment by AustinT** [comment]
> I am glad to receive your reply, but I tried the first method and found that after saving as a pkl file, I cannot transfer the file to my local computer. It shows 'can not create local file', which is something I cannot solve.

Transferring files to your local computer is not something I can help with unfortunately. I would personally use `scp` or `rsync` to move files. I suggest you search for documentation for these programs to try to resolve your issue. Good luck!

**Comment by TuTuSklmq** [comment]
I'm tempted to bother you again, I found that there was an error when reading the pkl file that caused my kernel to crash,&nbsp;I tried changing the part of the target.py file about the dock, but the output didn't change anything after changing the file
I would appreciate it if you would be willing to help me, so I'm asking for your help if there's a way to save and export the pdb file to the path I want.
A request for help from a recent college student


ROSA&nbsp;&nbsp;&nbsp;MULTIFLORA
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dockstring/dockstring"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月21日(星期四) 晚上6:11
***@***.***&gt;;
抄送:&nbsp;"ROSA   ***@***.******@***.***&gt;;
主题:&nbsp;Re: [dockstring/dockstring] Can we save the composite protein after molecular docking instead of viewing it directly? (Issue #170)





 
Closed #170 as completed.

**Comment by AustinT** [comment]
> so I'm asking for your help if there's a way to save and export the pdb file to the path I want. A request for help from a recent college student

@TuTuSklmq I can't really provide support for issues not related to dockstring, I'm sorry.

Also, to be perfectly honest, I think your request is a little bit selfish. If you think for just a moment about what you are asking me, you obviously have not provided enough detail for me to help you. I don't even know what OS you are using, much less what commands you actually tried and what errors you received. How exactly do you expect me to help you? Should I ask you bit by bit for information to slowly narrow down your issue? I don't have time for that. Generally, if you want free help from people on the internet, you should provide all the information up front to make it easy for people to help you...
